src/components/Pagination/Pagination.tsx

```diff
@@ -72,34 +71,43 @@ export const Pagination: FC<PaginationProps> = ({
   numberOfPagesDisplayed = 5,
   prevPageText = 'Previous',
 }) => {
-  const pageTotal = useMemo(
-    () => generatePageTotal(totalItemsCount, itemsPerPage),
-    [totalItemsCount, itemsPerPage]
-  );
+  const pageTotal = useMemo(() => {
+    if (itemsPerPage <= 0) return 1;
+    return generatePageTotal(totalItemsCount, itemsPerPage);
+  }, [totalItemsCount, itemsPerPage]);
 
-  const pageRange = useMemo(
-    () => generatePageRange(numberOfPagesDisplayed, pageTotal),
-    [numberOfPagesDisplayed, pageTotal]
-  );
+  const validActivePage = Math.max(1, Math.min(activePage, pageTotal));
 
   const activeListRange = useMemo(
-    () => generateActiveListRange(activePage, totalItemsCount, itemsPerPage),
-    [activePage, totalItemsCount, itemsPerPage]
+    () =>
+      generateActiveListRange(validActivePage, totalItemsCount, itemsPerPage),
+    [validActivePage, totalItemsCount, itemsPerPage]
   );
 
   const pages = useMemo(
-    () =>
-      generatePages(pageRange, pageTotal, activePage, numberOfPagesDisplayed),
-    [pageRange, pageTotal, activePage, numberOfPagesDisplayed]
+    () => generatePages(pageTotal, validActivePage, numberOfPagesDisplayed),
+    [pageTotal, validActivePage, numberOfPagesDisplayed]
+  );
+
+  const paginationClassNames = useMemo(
+    () => classNames(className),
+    [className]
   );
```

Comment:
Dynamically generates CSS class names if a custom class is provided, using `useMemo `to avoid unnecessary re-renders